### DISCUSSION TITLE
Help with moving entries

### DISCUSSION THREAD

**Discussion by Slim08** [moudey/Shell#578] [Q&A] [answered]
I'm trying to move all AV entries and all archiving entries each to a single submenu, 'Scan_with...' and 'Archivieren...'. And then move these submenus to 'more_options'. I can't figure out how to word this correctly. Any help would be appreciated.

```
modify(mode=mode.multiple( 
find='VirusTotal',
find='Microsoft Defender',
find='Malwarebytes'
)
pos=2 menu=title.Scan_with...)
```

```
modify(mode=mode.multiple
where=title.id(
id.7-Zip,
id.WinRar
)
pos=3 menu=title.Archivieren...)
```

**Comment by moudey** [comment, marked as answer]
```js
menu(mode="multiple" title="Scan with") {  }
modify(mode="multiple" find="VirusTotal|Microsoft Defender|Malwarebytes" pos=2 menu="Scan with")

```
```js

menu(mode="multiple" title="Archivieren") { }
modify(mode="multiple" find="7-Zip|WinRar" pos=3 menu="Archivieren")
```

**Reply by Slim08** [reply]
Thank you! And if I want to move those newly created submenus from the 'main' menu to the 'more_options' menu?

**Reply by ghost** [reply]
For moving the menus you created, either place them inside the braces of `More options` menu (this menu exists in `shell.nss`) :
```js
menu (mode="multiple" title=title.more_options type='~Recyclebin|~Taskbar' image=icon.more_options)
{
    menu(mode="multiple" title="Scan with") {  }
    menu(mode="multiple" title="Archivieren") { }
}
```
_OR_ specify their parent menu:
```js
menu(mode="multiple" title="Scan with" menu=title.more_options) { }
menu(mode="multiple" title="Archivieren" menu=title.more_options) { }
```
Finally for manipulating the system entries use `menu` property while separating each level of nesting with `/`:
```js
modify(mode="multiple" find="VirusTotal|Microsoft Defender|Malwarebytes" pos=2 menu='@title.more_options/Scan with')
modify(mode="multiple" find="7-Zip|WinRar" pos=3 menu='@title.more_options/Archivieren')
```

**Comment by Eta-Beta** [comment]
Good morning, I'm jumping into this discussion to ask a question. With these 2 command lines inserted in a customizations.nss file 

`menu(mode="multiple" title="Compattatori" image=\uE1E8){ }
Modify(type='file|dir' find='"7-zip"|"Estrai Tutto"' menu="Compattatori")`

I got this result

![Screenshot 2025-02-03 193716](https://github.com/user-attachments/assets/f304b5fc-fea1-4a71-88a5-1f89963258ea)

Unfortunately I have 2 problems, the first and most important is that the "Compactors" menu does not appear if I want to compress one or more image/video/audio files.... I have not checked if it happens for other files. The second less important but frustrating is that the 7-Zip icons do not appear. The result I would like to obtain is like in my old version 1.8.1, so

![Screenshot 2025-02-03 195946](https://github.com/user-attachments/assets/0653a271-057c-4b4b-a7d1-5c5214615c5d)


P.S. I don't know how to insert the strings in the post, I have to use "Quote?"

**Comment by RubicBG** [comment]
I'm a little confused by the explanations. I'll try to guess.
- If you create an empty menu and no system commands have been moved or cannot be moved, it will not be visible. So, do the `7-zip` and `Estrai Tutto` commands are moved when you select more than one file? Try with `mode='multiple'`:
```js
modify(mode="multiple" type='file|dir' find='"7-zip"|"Estrai Tutto"' menu="Compattatori")
```
- I don't know why there is no icon for `7-zip`, but try to add a new one with `modify()`:
```js
modify(mode="multiple" type='file|dir' find='"7-zip"' menu="Compattatori" image=path.combine(sys.prog,'7-Zip','7zFM.exe'))
modify(mode="multiple" type='file|dir' find='"Estrai Tutto"' menu="Compattatori")
```

**Comment by Eta-Beta** [comment]
Hi, first of all thanks for the reply, with the addition of this code:
image=path.combine(sys.prog,'7-Zip','7zFM.exe') now the icon appears. Unfortunately the main problem is not solved, I try to explain it again (I apologize but I use Google Translate to communicate and I understand that sometimes the meaning of the sentences can be unclear). When I select a folder with the right button the "Compactors" item appears and I can eventually compress it, if I select one or more files of type exe/dll/doc/excel....... almost all, the Compactors item appears, but with audio and video files of any format the Compactors item does not appear. I hope I explained myself a little better. Thanks.

UPDATE:
Stop everything I also installed WinRAR and now it also works on audio and video files ??????? maybe with just one voice it wasn't enough .... I'll never know, anyway I'm happy now I have my modification. Another question, in the old PC the Nilesoft skin is bright blue and I found it on the Nilesoft forum. To bring this code to version 1.9.18 how do I do it?. Thanks.

**Comment by Eta-Beta** [comment]
Hello, I have a new question regarding the movement of the created menus.

with this string I create the menu with a custom icon:

`menu(mode="multiple" title="Antivirus" image='C:\Programmi Portable\Icone Personali\virus 3D.ico') {}`

with this string I move it file options:

`menu(mode="multiple" title="Antivirus" menu=title.more_options) {}
`

with this string I insert the Antivirus menu items:

`modify(mode="multiple" find="Ricerca virus|Controlla reputerzione in KSN" pos=2 menu='@title.more_options/Antivirus')
`

This is the result and it works but there is no icon 😭😭

![Screenshot 2025-02-20 203041](https://github.com/user-attachments/assets/5d96c9df-f38b-4201-9929-aec82f5e49f8)

SORRY, SORRY. All it took was a moment to think about it, and the result was achieved.

`menu(mode="multiple" title="Antivirus" image='C:\Programmi Portable\Icone Personali\virus 3D.ico' menu=title.more_options) {}`

![Screenshot 2025-02-20 205514](https://github.com/user-attachments/assets/995c7aa6-0926-4918-9dac-043228c31b2e)
